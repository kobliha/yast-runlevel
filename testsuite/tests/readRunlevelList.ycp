/**
 * YaST2 test case for runlevel editor
 *
 * tests function readRunlevelList
 */
{
    include "runlevel/services.ycp";
    include "testsuite.ycp";

    map READ = $[
        "init" : $[
            "scripts" : $[
                "runlevel_list" : ["1", "2", "3"],
                ]
            ]
        ];

    list runlevels = [];
    map runlevel2tableindex = $[];

    TEST (``(readRunlevelList ()), [READ] , 0);
    DUMP (runlevels);
    DUMP (runlevel2tableindex);    

    READ = $[
        "init" : $[
            "scripts" : $[
                "runlevel_list" : [],
                ]
            ]
        ];

    TEST (``(readRunlevelList ()), [READ] , 0);
    DUMP (runlevels);
    DUMP (runlevel2tableindex);    
}
