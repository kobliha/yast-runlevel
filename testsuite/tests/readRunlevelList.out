Read	.init.scripts.runlevel_list ["1", "2", "3"]
Return	nil
Dump	["1", "2", "3"]
Dump	$["1":2, "2":3, "3":4]
Read	.init.scripts.runlevel_list []
Return	nil
Dump	["0", "1", "2", "3", "5", "6", "S"]
Dump	$["0":2, "1":3, "2":4, "3":5, "5":6, "6":7, "S":8]
